aten/src/ATen/native/mps/operations/Copy.mm

```diff
@@ -318,6 +318,41 @@ void copy_blit_mps(void* dst, const void* src, size_t size) {
   return dst_;
 }
 
+// tries to copy scalar buffers if any of them are allocated on MPSAllocator
+// with Shared-storage mode on Unified devices (supports CPU <-> MPS copies only).
+static bool try_copy_scalars_mps(at::Tensor& dst, const at::Tensor& src, bool non_blocking)
+{
+  const auto& allocator = *getIMPSAllocator();
+  if (!allocator.isSharedStorageSupported()) {
+    return false;
+  }
+  void* src_ptr = src.storage().data();
+  void* dst_ptr = dst.storage().data();
+  uint32_t retainCount = 0;
+
+  if (src.device().type() == at::kMPS) {
+    std::tie(src_ptr, retainCount) = allocator.getSharedBufferPtr(src_ptr);
+  }
+  if (dst.device().type() == at::kMPS) {
+    std::tie(dst_ptr, retainCount) = allocator.getSharedBufferPtr(dst_ptr);
+  }
+  if (!src_ptr || !dst_ptr) {
```

Comment:
>You already check if the cpu_ptr is NULL
Where? I don't check that in this function. 
>So you have a [buffer contents] in this case. 
If only the buffer was allocated on MPSAllocator and has Shared Storage mode this will be non-null.